### ISSUE TITLE
Support for pre-processors in .vuepress/components

### ISSUE THREAD

**Issue by aminimalanimal** [vuejs/vuepress#146]
Is there a way to get VuePress to support pre-processors like pug and sass?

**Comment by ulivz** [comment]
Good question, thanks! 

For using `pug` and `sass` at you custom component, you need to do following things:

1. Install essential dependencies:

```bash
yarn add sass-loader node-sass -D # for sass
yarn add pug-plain-loader pug -D # for pug
```

Since config for `scss`, `sass`, `stylus`, `less` has been built in VuePress, you ONLY need to extend for `pug` at your `.vuepress/config.js`:

```js
module.exports = {
  chainWebpack: config => {
    config.module
      .rule('pug')
      .test(/\.pug$/)
      .use('pug-plain-loader')
        .loader('pug-plain-loader')
        .end()
  }
}

```

Then you can write a custom component at `.vuepress/components`:

```vue
<template lang="pug">
  p.pug {{ msg }}
</template>

<script>
  export default {
    data () {
      return {
        msg: 'Hello, Pug'
      }
    }
  }
</script>

<style lang="sass">
  .pug
    font-size: 20px
</style>
```

Enjoy your writing!

**Comment by yyx990803** [comment]
We should add built-in config for pug and document this in the config section.

**Comment by aminimalanimal** [comment]
Wonderful! That worked.

I don't know if this is worth documenting as well, but I wasn't entirely sure whether I should install the dependencies within the `.vuepress` folder or (the correct way) from the root of my project, which makes `package.json`, `node_modules`, and `.vuepress` siblings.

**Comment by ulivz** [comment]
fixed at #151 and docs are here: https://vuepress.vuejs.org/config/#build-pipeline
